src/operator/batch_norm-inl.h

```diff
@@ -207,21 +212,26 @@ class BatchNormProp : public OperatorProperty {
     CHECK_EQ(in_shape->size(), 3U) << "Input:[data, gamma, beta]";
     const TShape &dshape = in_shape->at(0);
 
+    CHECK_GE(param_.channel_axis, -1) << "Invalid channel axis: " << param_.channel_axis;
```

Comment:
axis = -1 means first dim from right, axis=-2 means second dim from the right. You are only special casing -1 here. We need to support all